### DISCUSSION TITLE
How to call a query proactively?

### DISCUSSION THREAD

**Discussion by kimprosh** [trpc/trpc#5380] [Q&A] [answered]
In my case: search data when user input to search box. If the user don't input to search, don't call API to get data.
This action is getting data so I think I shouldn't use mutation but with query I don't have any idea to resolve it?

Any you guys have the solution?
Thanks!

**Comment by Nick-Lucas** [comment, marked as answer]
useState for the input value and useQuery with enabled=!!value

State drives everything in react so useQuery is designed around that

**Reply by kimprosh** [reply]
Thanks bro!
